### PR TITLE
hooks: add hook to run arbitrary commands

### PR DESCRIPTION
**Pull request by danielkza**

### CODE REVIEW

File: stacker/hooks/command.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        ignore_status (bool, optional):
+            Don't fail the hook if the command returns a non-zero status.
+            Default: false
+        quiet (bool, optional):
+            Redirect the command's stdout and stderr to the null device,
+            silencing all output. Should not be enaled if `capture` is also
+            enabled. Default: false
+        stdin (str, optional):
+            String to send to the stdin of the command. Implicitly disables
+            `intearctive`.
```
Comment (phobologic) [inline stacker/hooks/command.py]: typo

File: stacker/hooks/command.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+                  capture: true
+              - path: stacker.hooks.command.run_command
+                args:
+                  command: `cd $PROJECT_DIR/project; npm install'
+                  env:
+                    PROJECT_DIR: ./my-project
+                  shell: true
+    """
+
+    if quiet:
```
Comment (phobologic) [inline stacker/hooks/command.py]: Can we throw an error if quiet & capture are set?  Not 100% sure it's required, but might be nice and avoid some confusion later.

File: stacker/hooks/command.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        out_err_type = None
+
+    if interactive:
+        in_type = None
+    elif stdin:
+        in_type = PIPE
+    else:
+        in_type = _devnull()
+
+    if env is not None:
```
Comment (phobologic) [inline stacker/hooks/command.py]: `if env:` should be good enough here

File: stacker/hooks/command.py [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+        full_env = os.environ.copy()
+        full_env.update(env)
+        env = full_env
+
+    proc = Popen(command, stdin=in_type, stdout=out_err_type,
+                 stderr=out_err_type, env=env, **kwargs)
+    try:
+        out, err = proc.communicate(stdin)
```
Comment (phobologic) [inline stacker/hooks/command.py]: Probably worth a log statement here detailing the command being ran - probably debug?

**Comment by phobologic** [comment]
BTW, awesome set of tests :)

**Comment by phobologic** [comment]
ping - @danielkza I think @ejholmes is likely to release stacker 1.3 this week.  Just wanted to give you a heads up in case you want to get this in the next version.

**Comment by danielkza** [comment]
@phobologic Rebased and updated for your suggestions. Sorry for taking so long, I actually completely forget I had this PR open.

**Comment by phobologic** [comment]
@danielkza no worries, I figured :)  Just wanted to give you the heads up!

**Pull request merged**